### PR TITLE
`Development`: Reduce transferred data for course (un)enrollment endpoints

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

Currently the endpoints for enrollment and unenrollment return the complete User although only the updated Groups are required.

### Description

This PR changes the return type of the (un)enrollment endpoints to return only the set of groups.
The client services are updated accordingly.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Course with enrollment & unenrollment enabled

1. Log in to Artemis
2. Navigate to course enrollment
3. Make sure you can enroll in the course and access the content
4. In the course overview click unenroll from the course
5. Make sure you can unenroll from the course and you don't see it on the dashboard anymore

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [account.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5285/latest/artifact/shared/Coverage-Report-Client-Tests/app/core/auth/account.service.ts.html) | 87.62% | ✅ |
| [course-management.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5285/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/manage/course-management.service.ts.html) | 87.66% | ✅ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [CourseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5285/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/CourseResource.html) | 94% | ✅ |

### CODE REVIEW

File: src/test/javascript/spec/service/account.service.spec.ts [hunk 1]
```diff
@@ -108,7 +108,7 @@ describe('AccountService', () => {
     it('should sync user groups', () => {
         accountService.userIdentity = user;
 
-        accountService.syncGroups(user3);
+        accountService.syncGroups(user3.groups);
```
Comment (pal03377) [inline src/test/javascript/spec/service/account.service.spec.ts:111]: type error (which you already told me you will fix soon :) )

**Review by pal03377 — APPROVED** [review]
The code changes look like a good + clean performance improvement!

**Review by pal03377 — APPROVED** [review]
code

**Review by julian-christl — APPROVED** [review]
As you are not directly accessing the user resource, returning a user is not in any way useful.
Good job, code approved.

**Review by jakubriegel — CHANGES_REQUESTED** [review]
Local tests:
- I was not able to enrol for the course myself (the course didn't appear on the list
- but I was able to enrol for the course by admin
- I was able to unenroll myself

**Comment by MaximilianAnzinger** [comment]
> Local tests:
> 
> * I was not able to enrol for the course myself (the course didn't appear on the list
> * but I was able to enrol for the course by admin
> * I was able to unenroll myself

I'm not able to reproduce this. See the attached (Test course with (un)enrollment configured & logged in as a test user (1) )

Screenshot 2023-08-07 at 12 57 58

Is there a error message or something that might explain this?

**Review by sleiss — APPROVED** [review]

**Pull request merged**